packages/grafana-data/src/transformations/transformers/convertFieldType.ts

```diff
@@ -199,3 +221,32 @@ export function ensureTimeField(field: Field, dateFormat?: string): Field {
   }
   return fieldToTimeField(field, dateFormat);
 }
+
+/// Complex field parsing. Currently only JSON is supported, but more formats can be added
+/// by extending the ComplexFieldParserID enum and ComplexFieldParser interface.
+
+export enum ComplexFieldParserID {
+  JSON = 'json',
+}
+
+export interface ComplexFieldParser extends RegistryItem {
+  parse: (v: any) => any | null;
+}
+
+const jsonFieldParser: ComplexFieldParser = {
+  id: ComplexFieldParserID.JSON,
+  name: 'JSON',
+  description: 'Parse field as JSON',
+  parse: (v: any) => {
+    try {
+      return JSON.parse(v);
+    } catch {
+      return null;
+    }
+  },
+};
+
+/**
+ * Registry of complex field parsers, used with the ConvertFieldTypeTransformer.
+ */
+export const complexFieldParsers = new Registry<ComplexFieldParser>(() => [jsonFieldParser]);
```

Comment:
Do we need this abstraction?  I'm not sure what other things may land here -- and in particular should it just be the conversion type.  Perhaps the target should be 'Parse JSON' rather than 'other' -- we could extend the target type as needed -- but I'm not sure what else we need right now.

Reply:
Possibly not, I wasn't a big fan of equating `FieldType.other` with `JSON` but as long as we can change it later I guess it's fine.